includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -680,7 +680,7 @@ protected function update_version_and_type( &$product ) {
 	 */
 	protected function clear_caches( &$product ) {
 		wc_delete_product_transients( $product->get_id() );
-		wp_cache_delete( 'product-' . $product->get_id(), 'products' );
+		wp_cache_delete( 'object-' . $product->get_id(), 'products' );
```

Comment:
The order factory will now set an `object-` prefixed cache item and `clear_cache()` in the order data store will delete the item at an `object-` prefixed key. The product factory sets a `product-` prefixed cache item, and `cleare_cache()` in the product data store will delete the item at an `object-` prefixed key, which which won't do anything.